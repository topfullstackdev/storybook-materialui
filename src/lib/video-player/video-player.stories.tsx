import { Meta, Story } from '@storybook/react';

import { VideoPlayer } from './video-player';

export default {
  component: VideoPlayer,
  title: 'VideoPlayer',
  args: {}, // Common args
} as Meta<React.VideoHTMLAttributes<HTMLVideoElement>>;

function Template(props: React.VideoHTMLAttributes<HTMLVideoElement>) {
  return <VideoPlayer {...props} />;
}

export const Primary: Story<React.VideoHTMLAttributes<HTMLVideoElement>> =
  Template.bind({});
Primary.args = {
  src: 'http://commondatastorage.googleapis.com/gtv-videos-bucket/sample/ForBiggerBlazes.mp4',
  autoPlay: true,
  controls: true,
};
