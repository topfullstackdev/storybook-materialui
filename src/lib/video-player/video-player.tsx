import React from 'react';

export function VideoPlayer({
  children,
  ...props
}: React.VideoHTMLAttributes<HTMLVideoElement>) {
  return (
    <video {...props}>
      <track default kind="captions" />
      {children}
    </video>
  );
}

export default VideoPlayer;
