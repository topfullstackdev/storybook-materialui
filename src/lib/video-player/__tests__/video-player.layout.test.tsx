// Uses snapshot to test layout rendering
import { render } from '@testing-library/react';

import { VideoPlayer } from '../video-player';

describe('VideoPlayer', () => {
  it('should render successfully', () => {
    const { container } = render(<VideoPlayer />);
    expect(container).toMatchSnapshot();
  });
});
