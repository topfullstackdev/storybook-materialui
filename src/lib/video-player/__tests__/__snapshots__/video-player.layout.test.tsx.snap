// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer should render successfully 1`] = `
<div>
  <video>
    <track
      default=""
      kind="captions"
    />
  </video>
</div>
`;
